later/lib/database/datasources/records_local_data_source.dart

```diff
@@ -36,13 +35,49 @@ class RecordsLocalDataSource extends RecordsDataSource {
     return record;
   }
 
+  @override
+  Future<RecordDbModel> update(RecordDbModel recordDbModel) async {
+    final dbModel = await _isar.recordDbModels.get(recordDbModel.id);
+
+    if (dbModel != null) {
+      dbModel
+        ..title = recordDbModel.title == dbModel.title
+            ? dbModel.title
+            : recordDbModel.title
+        ..url =
+            recordDbModel.url == dbModel.url ? dbModel.url : recordDbModel.url
+        ..description = recordDbModel.description == dbModel.description
+            ? dbModel.description
+            : recordDbModel.description
+        ..lastEditedAt = DateTime.now();
+    } else {
+      throw const FormatException(
+          'We can not update file that we don\'t have at Database.');
+    }
```

Comment:
Can be refactored to this:

```dart
if (dbModel != null) {
  throw Exception('No entity with such ID found');
}

/* Go on and update ... */
```

By moving throwing the exception to be first, we reduces the amount of nesting.